haxe/ui/components/Button.hx

```diff
@@ -154,8 +154,8 @@ class ButtonLayout extends DefaultLayout {
             var label:Label = component.findComponent(Label, false);
             var ucx = usableSize.width;
             if (label != null) {
-                if (label.width > 0 && _component.componentWidth > 0 &&  label.width >= _component.componentWidth) {
-                    label.width = label.layout.calcAutoWidth();
+                if (label.width > 0 && _component.componentWidth > 0 && ucx > 0 && label.width >= ucx) {
+                    label.width =  ucx;
                 }
```

Comment:
Add the following so the label width "recovers" when dinamically (percent wise) we change the width of the button from lower (that has the text wrapped) to higher:

            else if (label.width > 0 && _component.componentWidth > 0 && ucx > 0) {
                label.width = label.layout.calcAutoWidth(); 
            }